dotnet/src/SemanticKernel.Abstractions/Contents/StreamingFunctionCallUpdateContent.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents a function streaming call requested by LLM.
+/// </summary>
+[Experimental("SKEXP0001")]
+public class StreamingFunctionCallUpdateContent : StreamingKernelContent
+{
+    /// <summary>
+    /// The function call ID that can come as full or partial.
+    /// </summary>
+    public string? Id { get; }
+
+    /// <summary>
+    /// The function name that can come as full or partial.
+    /// </summary>
+    public string? Name { get; }
+
+    /// <summary>
+    /// The function arguments that can come as full or partial.
+    /// </summary>
+    public string? Arguments { get; }
```

Comment:
I'm doubting if this is needed at all, in python we store the arguments as string in the FunctionCallContent class and then only when we want to use them do we parse to a dict/KernelArguments, and we have a update method (add) to combine mulitple items together for streaming, isn't that a bit simpler?

Reply:
As far as I understand, in SK python, `FunctionCallContent` is used for both streaming and non-streaming scenarios. In SK .NET, this is not the case.

SK .NET has two distinct hierarchies of content classes: one for streaming content, represented by the `StreamingKernelContent` base class, and the other for non-streaming content, represented by `KernelContent`. Content classes from each hierarchy are designed to represent not only the specifics of the content - such as image, audio, or function - but also the specifics of the way the content is communicated to API callers - streaming or non-streaming. Content classes belonging to the streaming hierarchy represent partial content, updates, or chunks that need to be combined together to create full content. Conversely, non-streaming classes represent fully baked content that doesn't need to be assembled together and can be used as is.

Back to the `FunctionCallContent`. The class inherits the `KernelContent` class and as such belongs to the non-streaming hierarchy and represents a fully-fledged function call. To use it for streaming purposes as well as for non-streaming ones, it would need to be inherited from the `StreamingKernelContent` base class as well. .NET doesn't allow multiple inheritance and as such it's not possible to have one class belonging to two hierarchies. 

Even if it were possible to have `FunctionCallContent` inherit two classes, it would have to be redesigned because its public `InvokeAsync` method would not work in a way callers might have expected if it was called on partial/streaming content.

Reply:
@SergeyMenshykh  thanks for the clarification, the big difference is the fact that in python we can inherit from multiple classes, so we have chosen long ago to base everything off `KernelContent` and we created the `StreamingContentMixin` to create the classes that come from streaming api's. This makes sure we have similar behavior when looking at something like StreamingTextContent vs TextContent and allows our type checking code to compare something to TextContent where it can then be either Streaming or not. So good to understand why we have this difference and also means there is no need at this time to add StreamingFunctionCallContent to Python.